src/modm/driver/radio/dw3110/dw3110_phy_impl.hpp

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2024, Elias H.
+ * Copyright (c) 2024, Raphael Lehmann
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef MODM_DW3110_PHY_HPP
+#error "Don't include this file directly, use 'dw3110_phy.hpp' instead!"
+#endif
+
+#include <modm/debug/logger.hpp>
+#include <modm/math/utils/endianness.hpp>
+
+#include "dw3110_definitions.hpp"
+#include "dw3110_phy.hpp"
+
+template<typename SpiMaster, typename Cs, uint32_t buffer_size>
+modm::ResumableResult<void>
+modm::Dw3110Phy<SpiMaster, Cs, buffer_size>::initialize()
+{
+	RF_BEGIN();
+	Cs::setOutput(true);
+	RF_END();
+}
+
+template<typename SpiMaster, typename Cs, uint32_t buffer_size>
+modm::ResumableResult<bool>
+modm::Dw3110Phy<SpiMaster, Cs, buffer_size>::ping()
+{
+	RF_BEGIN();
+	RF_CALL_BLOCKING(send<modm::Dw3110Register::SpiMode::FULL_READ>(
+		modm::Dw3110Register::DwRegister::DEV_ID, ping_result));
+	RF_END_RETURN((*((uint32_t *)ping_result.data()) & ~(0b1111)) == 0xDECA0300);
+}
+
+template<typename SpiMaster, typename Cs, uint32_t buffer_size>
+template<modm::Dw3110Register::SpiMode mode>
+modm::ResumableResult<uint32_t>
+modm::Dw3110Phy<SpiMaster, Cs, buffer_size>::send(modm::Dw3110Register::DwRegister address)
+{
+	RF_BEGIN();
+	RF_END_RETURN_CALL(send<mode>(address, std::span<uint8_t>()));
```

Comment:
This can be abbreviated: 

```suggestion
	return send<mode>(address, std::span<uint8_t>());
```

(But I don't understand what the inplace created `std::span` is meant to do.)